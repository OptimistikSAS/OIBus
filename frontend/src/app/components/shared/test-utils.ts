import { Provider } from '@angular/core';
import { NgbConfig } from '@ng-bootstrap/ng-bootstrap';

const NO_ANIMATION_NGB_CONFIG: NgbConfig = { animation: false };

export const noAnimation: Provider = { provide: NgbConfig, useValue: NO_ANIMATION_NGB_CONFIG };
