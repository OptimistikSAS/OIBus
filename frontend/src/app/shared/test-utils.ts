import { Provider } from '@angular/core';
import { NgbConfig } from '@ng-bootstrap/ng-bootstrap';
import { Page } from './types';

const NO_ANIMATION_NGB_CONFIG: NgbConfig = { animation: false };

export const noAnimation: Provider = { provide: NgbConfig, useValue: NO_ANIMATION_NGB_CONFIG };

export function toPage<T>(content: Array<T>, totalElements: number = content.length, num = 0, size = Math.max(20, totalElements)): Page<T> {
  return {
    content,
    totalElements,
    number: num,
    size,
    totalPages: Math.ceil(totalElements / size)
  };
}

export function emptyPage<T>(size = 20): Page<T> {
  return {
    totalElements: 0,
    content: [] as Array<T>,
    size,
    totalPages: 0,
    number: 0
  };
}
