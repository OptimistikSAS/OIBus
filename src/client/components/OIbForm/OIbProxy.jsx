import React from 'react'
import PropTypes from 'prop-types'
import { FormGroup, FormText, Label, Input } from 'reactstrap'
import { ConfigContext } from '../../context/configContext.jsx'

const OIbProxy = ({ label, help, value, name, onChange }) => {
  const { newConfig } = React.useContext(ConfigContext)
  const proxies = newConfig?.engine?.proxies ?? [] // proxies defined in engine
  const options = proxies.map((e) => e.name)
  options.unshift('') // allow the user to select no proxy(empty string)
  const defaultValue = options[0]

  React.useEffect(() => {
    if (value === null) onChange(name, defaultValue)
  }, [value])

  const handleChange = (event) => {
    const { target } = event
    const { value: newVal } = target
    onChange(name, newVal, null)
  }
  // if value is null, no need to render
  if (value === null) return null
  return (
    <FormGroup>
      {label && <Label for={name}>{label}</Label>}
      <Input className="oi-form-input" type="select" id={name} name={name} onChange={handleChange} value={value}>
        {options.map((o) => (
          <option key={o} value={o}>
            {o}
          </option>
        ))}
      </Input>
      {help && <FormText>{help}</FormText>}
    </FormGroup>
  )
}
OIbProxy.propTypes = {
  name: PropTypes.string.isRequired,
  label: PropTypes.string,
  help: PropTypes.element,
  onChange: PropTypes.func.isRequired,
  value: PropTypes.string,
}
OIbProxy.defaultProps = { label: null, help: null, value: null }

export default OIbProxy
