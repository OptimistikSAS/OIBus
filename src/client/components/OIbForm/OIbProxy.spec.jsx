import React from 'react'
import ReactDOM from 'react-dom'
import { act } from 'react-dom/test-utils'
import newConfig from '../../../../tests/testConfig'
import OIbProxy from './OIbProxy.jsx'

React.useContext = jest.fn().mockReturnValue({ newConfig })

let container
beforeEach(() => {
  container = document.createElement('div')
  document.body.appendChild(container)
})

afterEach(() => {
  document.body.removeChild(container)
  container = null
})

describe('OIbProxy', () => {
  test('check Proxy with value="proxy"', () => {
    act(() => {
      ReactDOM.render(<OIbProxy
        label="label"
        value="proxy"
        name="name"
        onChange={() => (1)}
      />, container)
    })
    expect(container).toMatchSnapshot()
  })
  test('check Proxy with value="proxy" and no label', () => {
    act(() => {
      ReactDOM.render(<OIbProxy
        value="some text"
        name="name"
        onChange={() => (1)}
      />, container)
    })
    expect(container).toMatchSnapshot()
  })
  test('check Proxy with value and help text', () => {
    act(() => {
      ReactDOM.render(<OIbProxy
        label="label"
        value="proxy"
        help={<div>some help text</div>}
        name="name"
        onChange={() => (1)}
      />, container)
    })
    expect(container).toMatchSnapshot()
  })
})
