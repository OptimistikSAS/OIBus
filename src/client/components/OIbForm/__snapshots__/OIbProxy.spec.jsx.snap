// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OIbProxy check Proxy with value and help text 1`] = `
<div>
  <div
    class="form-group"
  >
    <label
      class=""
      for="name"
    >
      label
    </label>
    <select
      class="oi-form-input form-control"
      id="name"
      name="name"
    >
      <option
        value=""
      />
      <option
        value="sss"
      >
        sss
      </option>
      <option
        value="ff"
      >
        ff
      </option>
    </select>
    <small
      class="form-text text-muted"
    >
      <div>
        some help text
      </div>
    </small>
  </div>
</div>
`;

exports[`OIbProxy check Proxy with value="proxy" 1`] = `
<div>
  <div
    class="form-group"
  >
    <label
      class=""
      for="name"
    >
      label
    </label>
    <select
      class="oi-form-input form-control"
      id="name"
      name="name"
    >
      <option
        value=""
      />
      <option
        value="sss"
      >
        sss
      </option>
      <option
        value="ff"
      >
        ff
      </option>
    </select>
  </div>
</div>
`;

exports[`OIbProxy check Proxy with value="proxy" and no label 1`] = `
<div>
  <div
    class="form-group"
  >
    <select
      class="oi-form-input form-control"
      id="name"
      name="name"
    >
      <option
        value=""
      />
      <option
        value="sss"
      >
        sss
      </option>
      <option
        value="ff"
      >
        ff
      </option>
    </select>
  </div>
</div>
`;
